applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -591,13 +576,6 @@ static bool handle_ble_peer_operation_event(const struct ble_peer_operation_even
 	case PEER_OPERATION_ERASED:
 		reset_subscribers();
 		store_subscribed_peers();
-		if (count_conn() == CONFIG_BT_MAX_CONN) {
-			if (IS_ENABLED(CONFIG_BT_SCAN_CONN_ATTEMPTS_FILTER)) {
-				bt_scan_conn_attempts_filter_clear();
-			}
-			peers_only = false;
-			break;
-		}
```

Comment:
Is this no longer needed?

Reply:
+1

Reply:
There is no break, the fall-through causes executing the same code located below. The `scan_start` function would not start canning, because of internal conn count check.

Reply:
well actually there is if the count_conn() is equal to CONFIG_BT_MAX_CONN. In that case, the scan_start is not called

Reply:
Yes, but if the `count_conn() == CONFIG_BT_MAX_CONN`, the `scan_start` would not start scanning anyway and behavior is actually unchanged.

Reply:
okay, seems good to me. Thanks for explaining this simplification

I also appreciate the multi-commit structure of this PR which made it more review-friendly

Reply:
Okay, thanks for the explanation.